runtime/pkg/duration/duration.go

```diff
@@ -5,12 +5,20 @@ import (
 	"fmt"
 	"regexp"
 	"strconv"
+	"strings"
 	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 )
 
-// Duration represents an ISO8601 duration with Rill-specific extensions.
+type Duration interface {
+	Add(t time.Time) time.Time
+	Sub(t time.Time) time.Time
+}
+
+// StandardDuration represents an ISO8601 duration with Rill-specific extensions.
 // See ParseISO8601 for details.
-type Duration struct {
+type StandardDuration struct {
```

Comment:
The word "standard" kind of implies that it doesn't have Rill-specific extensions. The only Rill-specific extension it has is support for `inf`. With the new interface, would suggest moving that out into a separate type `InfDuration` (then this type becomes standard-only).